### PR TITLE
CRIMRE-45-sort-applications

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

CRIMRE-144 sort my list
CRIMRE-145 sort open applications
CRIMRE-146 sort closed applications

## Link to relevant ticket

[CRIMRE-144](https://dsdmoj.atlassian.net/browse/CRIMRE-144)
[CRIMRE-145](https://dsdmoj.atlassian.net/browse/CRIMRE-145)
[CRIMRE-146](https://dsdmoj.atlassian.net/browse/CRIMRE-146)

### After changes:
Screenshot 2023-01-30 at 17 15 38
Screenshot 2023-01-30 at 17 15 22
Screenshot 2023-01-30 at 17 15 16
Screenshot 2023-01-30 at 17 17 44

## How to manually test the feature

[CRIMRE-144]: https://dsdmoj.atlassian.net/browse/CRIMRE-144?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMRE-145]: https://dsdmoj.atlassian.net/browse/CRIMRE-145?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMRE-146]: https://dsdmoj.atlassian.net/browse/CRIMRE-146?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/views/application_searches/show.html.erb [hunk 1]
```diff
@@ -18,5 +18,31 @@
     </div>
   </div>
 
-  <%= render 'search_results_table', results: @search.results, form_params: { sorting: @sorting, filter: @filter, url: application_searches_path }  %>
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full">
+      <table class="govuk-table app-dashboard-table" data-module="moj-sortable-table">
+        <thead class="govuk-table__head">
+          <tr class="govuk-table__row">
+            <%= table_headers(
```
Comment (willmcb) [inline app/views/application_searches/show.html.erb:26]: Why not use the partial like in `app/views/application_searches/_search_results_table.html.erb`?

File: app/views/assigned_applications/index.html.erb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-full govuk-!-margin-bottom-1">
     <h1 class="govuk-heading-xl govuk-!-margin-bottom-1"><%=t '.heading' %></h1>
-    <p><%=t '.your_assignments', count: @applications.count %></p>
+    <p><%=t '.your_assignments', count: @search.total %></p>
```
Comment (willmcb) [inline app/views/assigned_applications/index.html.erb:6]: Space needed between the `=t` on these two lines 👍

File: app/views/assigned_applications/index.html.erb [hunk 3]
```diff
@@ -23,6 +23,30 @@
...
 
-<% if @applications.any? %>
-  <%= render 'application_table', applications: @applications %>
+<% unless @search.results.empty? %>
+  <div class="govuk-grid-row">
+      <div class="govuk-grid-column-full">
+        <table class="govuk-table app-dashboard-table">
+          <thead class="govuk-table__head">
+            <tr class="govuk-table__row">
+              <%= table_headers(
```
Comment (willmcb) [inline app/views/assigned_applications/index.html.erb:32]: I think we either use the partial or the helper - either one is fine but I'm not sure we need the partial if we are using it in some places and not others 🤔

File: spec/system/open_applications_dashboard_spec.rb [hunk 4]
```diff
@@ -15,7 +15,7 @@
   it 'includes the correct headings' do
     column_headings = page.first('.app-dashboard-table thead tr')
 
-    expect(column_headings.text).to eq('Applicant Ref. no. Date received Days passed Common Platform Caseworker')
+    expect(column_headings.text.squish).to eq('Applicant Ref. no. Date received Days passed Common Platform Caseworker')
```
Comment (willmcb) [inline spec/system/open_applications_dashboard_spec.rb:18]: Maybe move `.text.squish` to the variable assignment like in the other spec 👍

File: spec/system/searching/filter_by_date_range_spec.rb [hunk 5]
```diff
@@ -10,8 +10,8 @@
     visit '/'
     click_link 'Search'
 
-    fill_in 'search-application-search-filter-submitted-after-field', with: after_date
-    fill_in 'search-application-search-filter-submitted-before-field', with: before_date
+    fill_in 'filter-submitted-after-field', with: after_date
```
Comment (willmcb) [inline spec/system/searching/filter_by_date_range_spec.rb:13]: I line the pattern of having these in `lets` its nice to have consistancy across the specs if we can.

**Review by willmcb — COMMENTED** [review]
Really nice stuff - I have made a few comments - nothing big :) 

Lets chat through them in the morning and we can get this merged 👍

File: app/controllers/assigned_applications_controller.rb [hunk 6]
```diff
@@ -1,9 +1,11 @@
 class AssignedApplicationsController < ApplicationController
   def index
+    sort_params = params[:sorting]&.permit(Sorting.attribute_names)
```
[the remark this answers is not in the document]
Comment (timpeat) [reply app/controllers/assigned_applications_controller.rb:3]: They do depend on the context a bit. But there's definitely scope to extract some of the sorting code.

**Review by willmcb — APPROVED** [review]
Brill 👍

**Pull request merged**